app/src/main/java/ayds/spotisong/aurora/domain/otherdetails/FindInformationImp.java

```diff
@@ -0,0 +1,24 @@
+package ayds.spotisong.aurora.domain.otherdetails;
+
+import ayds.spotisong.aurora.presentation.otherdetails.model.OtherDetailsUseCaseInteractor;
+import ayds.spotisong.aurora.domain.song.Song;
+
+class FindInformationImp implements FindInformation {
+
+    private CardRepository repository;
+    private OtherDetailsUseCaseInteractor useCaseInteractor;
+
+    FindInformationImp(CardRepository repository){
+        this.repository = repository;
+    }
+
+    @Override
+    public void execute(Song song) {
+        repository.getCards(song, useCaseInteractor::onCardsFound, useCaseInteractor::onError);
```

Comment:
no hace falta q el caso de uso conozca al interactor de presenter. Se soluciona agregando los callbacks como parámetros:
```suggestion
public void execute(Song song, SearchListener listener, SearchErrorListener errorListener) {
        repository.getCards(song, listener, errorListener);
```